vscode/src/player/Subtitles.ts

```diff
@@ -0,0 +1,151 @@
+import * as vscode from 'vscode';
+import * as parser from 'subtitles-parser-vtt';
+
+/**
+ * Cue object properties.
+ */
+interface Cue {
+  id: number;
+  startTime: number;
+  endTime: number;
+  text: string;
+}
+
+/**
+ * Time unit in milliseconds to restrict given values to.
+ */
+const TIME_UNIT: number = 1000;
+
+/**
+ * Adjustment multiplier for timers to execute sooner than later.
+ */
+const ADJ_MUL: number = 0.98;
+
+/**
+ * Class handles loading and manipulation of the subtitles.
+ */
+export default class Subtitles {
+  private oc: vscode.OutputChannel;
+  private cuesDict: Object = {};
+  private startTimeUnit: number;
+  private timerRef: NodeJS.Timer;
+
+  /**
+   * Create Output Channel.
+   */
+  constructor() {
+    this.oc = vscode.window.createOutputChannel('Codio Subtitles');
+  }
+
+  /**
+   * Load given subtitles file and create dictionary.
+   * @param filePath Subtitles file to load.
+   */
+  async load(filePath: string): Promise<void> {
+    const uri = vscode.Uri.file(filePath);
+
+    try {
+      const srtData = (await vscode.workspace.openTextDocument(uri)).getText();
+      const cues = parser.fromSrt(srtData, 'ms');
+      if (!cues.length) {
+        throw new Error('No subtitles found.');
+      }
+
+      this.createCuesDict(cues);
+      this.display('Codio Subtitles Loaded');
+    } catch (error) {
+      console.error('Subtitles Load', error.message);
+      this.play = () => { };
```

Comment:
Interesting approaching. I wonder if it is better to handle the error state at the play method it self. Overriding a function feels like an anti pattern but this is also a very succinct way of solving the current problem. 
What are the tradeoffs?

Reply:
I agree about the code smell and I went back and forth on how to probably handle this. My view was that subtitles are a nice to have and aren't necessary for the player to work as it's main responsibility is the Editor and Audio. However, silently failing isn't always the best option as it is hidden from the owner.

Upon review, we can return a Boolean from load and then let the Player call Subtitles.destroy method on false which will override the Subtitles play method. This explicitly shows how the owner handles its dependency on a load error.

Reply:
I think this is indeed a better solution